License2Deploy/rolling_deploy.py

```diff
@@ -353,15 +364,16 @@ def deploy(self): # pragma: no cover
     group_name = self.get_autoscale_group_name()
     self.wait_ami_availability(self.ami_id)
     logging.info("Build #: {0} ::: Autoscale Group: {1}".format(self.build_number, group_name))
-    self.existing_instance_ids = list(self.get_all_instance_ids(group_name))
-    self.log_instances_ips(self.existing_instance_ids, group_name)
+    self.original_instance_ids = list(self.get_all_instance_ids(group_name))
+    self.log_instances_ips(self.original_instance_ids, group_name)
     if not self.force_redeploy and self.is_redeploy():
       self.stop_deploy('You are attempting to redeploy the same build. Please pass the force_redeploy flag if a redeploy is desired')
     self.disable_project_cloudwatch_alarms()
     self.new_desired_capacity = self.calculate_autoscale_desired_instance_count(group_name, 'increase')
     self.set_autoscale_instance_desired_count(self.new_desired_capacity, group_name)
     self.launch_new_instances(group_name)
-    self.set_autoscale_instance_desired_count(self.calculate_autoscale_desired_instance_count(group_name, 'decrease'), group_name)
+    self.terminate_original_instances(group_name)
+    self.set_autoscale_instance_desired_count(len(self.original_instance_ids), group_name)
```

Comment:
Is it possible that len(self.original_instance_ids) is different from calculate_autoscale_desired_instance_count() ?

Reply:
yes, it's actually likely that it will be different.

self.terminate_original_instances() has decrement_capacity=True so it will decrement each time, using len(self.original_instance_ids) will ensure that we finish with the same number of servers that we started with. this is to account for the edge case where self.terminate_original_instances() does not terminate successfully.

for example: If we started with 2 servers, then self.new_desired_capacity will be 4 servers. self.terminate_original_instances will take us back down to 2 servers. If we then ran self.calculate_autoscale_desired_instance_count(group_name, 'decrease') we would end at 1 server while len(self.original_instance_ids) will still end at 2.

Reply:
👍